src/mlpack/core/data/load_impl.hpp

```diff
@@ -164,7 +164,12 @@ bool Load(const std::string& filename,
   if (loadType != FileType::HDF5Binary)
   {
     if (loadType == FileType::CSVASCII)
-      success = loader.LoadNumericCSV(matrix, stream);
+    {
+      arma::field<std::string> headers;
+      success = matrix.load(arma::csv_name(filename, headers, arma::csv_opts::trans),
+          arma::csv_ascii);
+      transpose = false;
```

Comment:
This won't work if the user specified `transpose = false` to begin with---we should omit the option `arma::csv_opts::trans` if `transpose` is `false`.

Down below the `if (transpose)` can be replaced with `if (transpose && loadType == FileType::CSVASCII)`.